src/app/Plans/components/PlansTable.tsx

```diff
@@ -196,47 +196,14 @@ const PlansTable: React.FunctionComponent<IPlansTableProps> = ({
   const rows: IRow[] = [];
 
   currentPageItems.forEach((plan: IPlan) => {
-    let buttonType: PlanActionButtonType | null = null;
-    let title = '';
-    let variant: ProgressVariant | undefined;
-
-    const conditions = plan.status?.conditions || [];
     const latestMigration = findLatestMigration(plan, migrationsQuery.data?.items || null);
+    const isWarmPlan = plan.spec.warm;
 
-    // TODO This whole if-else pile should instead be reduced to a union type like WarmPlanState but generalized.
-    // TODO the PlanStatusType should be a union PlanConditionType and the PlanStatusDisplayType should perhaps not be a thing.
+    // TODO the PlanStatusDisplayType should perhaps not be a thing.
     // TODO note, buttonType and title will only be used if <Progress> is shown in the row below, so their values don't matter for pre-cutover.
-    if (hasCondition(conditions, PlanStatusType.Ready) && !plan.status?.migration?.started) {
-      buttonType = 'Start';
-    } else if (hasCondition(conditions, PlanStatusType.Executing)) {
-      buttonType = null;
-      title = PlanStatusDisplayType.Executing;
-      if (plan.spec.warm) {
-        title = 'Running cutover';
-        if (!latestMigration?.spec.cutover) {
-          buttonType = 'Cutover';
-        }
-      }
-    } else if (hasCondition(conditions, PlanStatusType.Succeeded)) {
-      title = PlanStatusDisplayType.Succeeded;
-      variant = ProgressVariant.success;
-    } else if (hasCondition(conditions, PlanStatusType.Canceled)) {
-      title = PlanStatusDisplayType.Canceled;
-    } else if (hasCondition(conditions, PlanStatusType.Failed)) {
-      title = PlanStatusDisplayType.Failed;
-      variant = ProgressVariant.danger;
-    } else if (plan.status?.migration?.started) {
-      if (canPlanBeStarted(plan)) {
-        title = 'Finished - Incomplete';
-        variant = ProgressVariant.warning;
-      } else {
-        console.warn('Migration plan unexpected status:', plan);
-      }
-    }
-
-    if (buttonType !== 'Start' && canPlanBeStarted(plan)) {
-      buttonType = 'Restart';
-    }
+    const planState = getPlanState(plan, latestMigration, migrationsQuery);
+    const newButtonType = getButtonState(planState, plan, latestMigration);
```

Comment:
I think we can call this `buttonType` since there won't be an old and a new one anymore :)

Reply:
lol whoops, leftovers from debug mode - will update